litellm/caching/redis_cache.py

```diff
@@ -136,12 +136,13 @@ def init_async_client(
             connection_pool=self.async_redis_conn_pool, **self.redis_kwargs
         )
 
-    def check_and_fix_namespace(self, key: str) -> str:
+    def check_and_fix_namespace(self, key: str, **kwargs) -> str:
         """
         Make sure each key starts with the given namespace
         """
-        if self.namespace is not None and not key.startswith(self.namespace):
-            key = self.namespace + ":" + key
+        namespace = kwargs.get("metadata", {}).get("redis_namespace") or kwargs.get("extra_body", {}).get("cache", {}).get("namespace") or self.namespace
+        if namespace is not None and not ":" in key:
```

Comment:
switched this check to check if `:` is in the key. Since, the hashed key is only hexadecimal, then a colon being present in the key means that it is prefixed with the namespace